src/Behat/Behat/Output/ServiceContainer/Formatter/JSONFormatterFactory.php

```diff
@@ -0,0 +1,190 @@
+            new Reference('output.json.statistics.feature'),
+            new Reference('output.node.listener.json.duration'),
+        ]);
+        $container->setDefinition('output.node.printer.json.feature', $definition);
+
+        $definition = new Definition(JSONScenarioPrinter::class, [
+            new Reference(self::RESULT_TO_STRING_CONVERTER_ID),
+            new Reference('output.node.listener.json.duration'),
+        ]);
+        $container->setDefinition('output.node.printer.json.scenario', $definition);
+
+        $definition = new Definition(JSONStepPrinter::class, [
+            new Reference(ExceptionExtension::PRESENTER_ID),
+        ]);
+        $container->setDefinition('output.node.printer.json.step', $definition);
+
+        $definition = new Definition(
+            JSONSetupPrinter::class,
+            [
+                new Reference(ExceptionExtension::PRESENTER_ID),
+            ]
+        );
+        $container->setDefinition('output.node.printer.json.setup', $definition);
+    }
+
+    private function loadRootNodeListener(ContainerBuilder $container): void
+    {
+        $definition = new Definition(
+            JSONDurationListener::class
+        );
+
+        $container->setDefinition('output.node.listener.json.duration', $definition);
+
+        $definition = new Definition(ChainEventListener::class, [
+            [
+                new Reference('output.node.listener.json.duration'),
+                new Definition(JSONElementListener::class, [
+                    new Reference('output.node.printer.json.exercise'),
+                    new Reference('output.node.printer.json.suite'),
+                    new Reference('output.node.printer.json.feature'),
+                    new Reference('output.node.printer.json.scenario'),
+                    new Reference('output.node.printer.json.step'),
+                    new Reference('output.node.printer.json.setup'),
+                ]),
+            ],
+        ]);
+        $container->setDefinition(self::ROOT_LISTENER_ID, $definition);
+    }
+
+    private function loadFormatter(ContainerBuilder $container): void
+    {
+        $definition = new Definition(PhaseStatistics::class);
+        $container->setDefinition('output.json.statistics.exercise', $definition);
+
+        $definition = new Definition(PhaseStatistics::class);
+        $container->setDefinition('output.json.statistics.suite', $definition);
+
+        $definition = new Definition(PhaseStatistics::class);
+        $container->setDefinition('output.json.statistics.feature', $definition);
+
+        $definition = new Definition(NodeEventListeningFormatter::class, [
+            JSONFormatter::NAME,
+            'Outputs the failures in JSON files.',
+            JSONFormatter::defaults(),
+            $this->createOutputPrinterDefinition(),
+            new Definition(ChainEventListener::class, [
+                [
+                    new Reference(self::ROOT_LISTENER_ID),
```

Comment:
We need to call the ScenarioStatsListener, StepStatsListener and HookStatsListener three times because we have three sets of statistics: per feature, per suite and per exercise